gemini-core/src/main/java/com/haroncode/gemini/connector/BaseStoreConnector.kt

```diff
@@ -0,0 +1,40 @@
+package com.haroncode.gemini.connector
+
+import com.haroncode.gemini.connection.ConnectionRule
+import io.reactivex.disposables.CompositeDisposable
+
+/**
+ * @author kdk96.
+ * @author HaronCode.
+ */
+open class BaseStoreConnector(
+    private val connectionRules: Collection<ConnectionRule>
+) : StoreConnector {
+
+    private val connectionRuleDisposables = CompositeDisposable()
+    private val retainRuleDisposables = CompositeDisposable()
+    private val lifecycleDisposables = CompositeDisposable(connectionRuleDisposables, retainRuleDisposables)
+
+    private var isNotConnectedRetain = true
```

Comment:
this variable isn't needed, cause we can check `retainRuleDisposables` size

Reply:
But if we don't have any retain connection we will try to connect retain connections every time